### DISCUSSION TITLE
What is OWIN? How does it relate to Katana and Kestrel

### DISCUSSION THREAD

**Discussion by giscus** [PradeepLoganathan/pradeepl-blog#64] [BlogPosts]
# What is OWIN? How does it relate to Katana and Kestrel

The Open Web Interface for .NET (OWIN) is a standard that defines an interface between .NET servers and web applications. Katana is an implementation of OWIN by the .Net foundation. Kestrel is a lightweight cross-platform web server that supports .NET Core and runs on multiple platforms such as Linux, macOS, and Windows.

https://pradeepl.com/blog/owin-katana-kestrel/

**Comment by reid-klein** [comment]
Good job on this document it made it easy to share with others! I do have a couple of corrections/questions to point in regard to Kestrel:

1) ... "Kestrel isn’t a full-featured web server, it’s a small, fast web server geared toward serving dynamic content from ASP.NET Core. It is designed to run behind a proxy like IIS on windows or NGINX on Linux/MacOS , and should not be used to directly host ASP.NET Core or any other application in production environments. " ...

I think what you meant to say about Kestrel in production is: It can be a good idea to use a reverse proxy of some sort in front of Kestrel like NGINX, IIS, YARP, etc.? But not that Kestrel shouldn't be used in production... This is circumstantial.

2) "Kestrel is packaged as a NuGet library ..." - 

The NuGet packages for Kestrel have been deprecated, you should remove this reference from the document. It is now directly included in ASP.NETCore

Thanks Pradeep, 

Reid
3) HTTP2 is now enabled by default. I believe since dotnet core 3.1 and forward.

**Reply by PradeepLoganathan** [reply]
Hello @reid-klein ,
  Thanks for taking the time to create this issue. I really appreciate it. Technology moves fast and this blog post is showing its age. I will update it when i find the time. However, please feel free to create a pull request on this github repo with the changes and I will review , approve and merge it. I would really encourage you to update the information in this issue directly on the blog post. If you need help with creating the PR, I am happy to help you. I will look forward to your PR.
